src/query/service/tests/it/sql/planner/optimizer/eliminate_in_subquery_test.rs

```diff
@@ -0,0 +1,421 @@
+                │   └── limit: NONE
+                └── Scan
+                    ├── table: default.t2
+                    ├── filters: []
+                    ├── order by: []
+                    └── limit: NONE\n",
+        },
+    ]
+}
+
+fn create_table_plan(fixture: &TestFixture, format: &str) -> Vec<CreateTablePlan> {
+    vec![
+        CreateTablePlan {
+            create_option: CreateOption::Create,
+            tenant: fixture.default_tenant(),
+            catalog: fixture.default_catalog_name(),
+            database: "default".to_string(),
+            table: "t1".to_string(),
+            schema: TableSchemaRefExt::create(vec![
+                TableField::new(
+                    "a",
+                    TableDataType::Nullable(Box::new(TableDataType::Number(NumberDataType::Int32))),
+                ),
+                TableField::new("b", TableDataType::Number(NumberDataType::Int32)),
+                TableField::new("c", TableDataType::String),
+            ]),
+            engine: Engine::Fuse,
+            engine_options: Default::default(),
+            storage_params: None,
+            options: [
+                // database id is required for FUSE
+                (OPT_KEY_DATABASE_ID.to_owned(), "1".to_owned()),
+                (OPT_KEY_STORAGE_FORMAT.to_owned(), format.to_owned()),
+            ]
+            .into(),
+            field_comments: vec![],
+            as_select: None,
+            cluster_key: None,
+            inverted_indexes: None,
+            attached_columns: None,
+        },
+        CreateTablePlan {
+            create_option: CreateOption::Create,
+            tenant: fixture.default_tenant(),
+            catalog: fixture.default_catalog_name(),
+            database: "default".to_string(),
+            table: "t2".to_string(),
+            schema: TableSchemaRefExt::create(vec![
+                TableField::new(
+                    "a",
+                    TableDataType::Nullable(Box::new(TableDataType::Number(NumberDataType::Int32))),
+                ),
+                TableField::new("b", TableDataType::Number(NumberDataType::Int32)),
+                TableField::new("c", TableDataType::String),
+            ]),
+            engine: Engine::Fuse,
+            engine_options: Default::default(),
+            storage_params: None,
+            options: [
+                // database id is required for FUSE
+                (OPT_KEY_DATABASE_ID.to_owned(), "1".to_owned()),
+                (OPT_KEY_STORAGE_FORMAT.to_owned(), format.to_owned()),
+            ]
+            .into(),
+            field_comments: vec![],
+            as_select: None,
+            cluster_key: None,
+            inverted_indexes: None,
+            attached_columns: None,
+        },
+    ]
+}
+
+#[tokio::test(flavor = "multi_thread")]
+async fn test_query_rewrite() -> Result<()> {
+    test_query_rewrite_impl("parquet").await?;
+    test_query_rewrite_impl("native").await
```

Comment:
don't need test 'native' format, as this feature is only used to optimize the query plan and has nothing to do with the storage layer.